src/libsyntax/ast.rs

```diff
@@ -1553,7 +1558,36 @@ pub enum ImplItemKind {
     Macro(Mac),
 }
 
-#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Hash, RustcEncodable, RustcDecodable, Copy)]
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash, RustcEncodable, RustcDecodable, Debug)]
+pub enum FloatTy {
+    F32,
+    F64,
+}
+
+impl FloatTy {
+    pub fn ty_to_string(self) -> &'static str {
```

Comment:
Why not implement `fmt::Display` as well? Also, the `ty_` prefix seems redundant here.

Reply:
May I suggest `name` or `ty_name` for the `Symbol` and `name_str` or `ty_name_str` for the `&'static str` form?

However, I should point out that a lot of types in the compiler are expected to implement `fmt::Display` for use in diagnostics and the method call looks like line noise by comparison.

Reply:
Renamed to `name` and `name_str`. As for `Display`, I've already removed a bunch of these from the AST before so that the dependencies are clearer.